src/lib/optimizer/strategy/like_replacement_rule.cpp

```diff
@@ -0,0 +1,91 @@
+#include "like_replacement_rule.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+#include "all_parameter_variant.hpp"
+#include "constant_mappings.hpp"
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "operators/operator_scan_predicate.hpp"
+#include "statistics/chunk_statistics/chunk_statistics.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/assert.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+std::string LikeReplacementRule::name() const { return "Like Replacement Rule"; }
+
+const MAX_ASCII_VALUE = 127;
```

Comment:
Brrr. Implicit int is... not nice. Please add a data type (or auto). Also, constexpr?